pilot/pkg/model/gateway.go

```diff
@@ -167,6 +168,13 @@ func MergeGateways(gateways []gatewayWithInstances) *MergedGateway {
 					if s.Tls.Mode == networking.ServerTLSSettings_AUTO_PASSTHROUGH {
 						autoPassthrough = true
 					}
+					if len(s.Tls.CipherSuites) > 0 {
+						serverName := s.Name
+						if len(serverName) == 0 {
+							serverName = s.Port.Name
```

Comment:
This is hacky

Reply:
So we either identify server with unique port name or server name. Why do you think it is hacky?